File: leetcode/2028-find-missing-observations/missingRolls.test.ts
import cases from "./cases.json";
import missingRolls from "./missingRolls";

test("find missing observations", () => {
  cases.forEach((testCase, index) => {
    console.log(`正在执行第 ${index + 1} 个测试用例\n`);
    const result = missingRolls(
      testCase.input.rolls,
      testCase.input.mean,
      testCase.input.n
    );
    console.log(
      `input: ${testCase.input} , output: ${testCase.output} ==> ${result} \n`
    );

    expect(result).toEqual(testCase.output);
  });
});

File: leetcode/2028-find-missing-observations/missingRolls_test.go
package leetcode

import (
	"fmt"
	"path/filepath"
	"reflect"
	"testing"

	"github.com/edram/leetcode/util"
)

type MissingRollsCase struct {
	Input struct {
		Rolls []int
		Mean  int
		N     int
	}

	Output []int
}

func TestMissingRolls(t *testing.T) {
	var casePath, _ = filepath.Abs("./cases.json")
	payload, err := util.LoadCases[[]MissingRollsCase](casePath)
	if err != nil {
		panic(err)
	}

	for i, testCase := range *payload {
		fmt.Printf("正在执行第 %d 个测试用例\n", i+1)
		fmt.Printf("string: %v ==> %v \n", testCase.Input, testCase.Output)

		result := missingRolls(testCase.Input.Rolls, testCase.Input.Mean, testCase.Input.N)
		if !reflect.DeepEqual(result, testCase.Output) {
			t.Errorf("期望 %v, 结果 %v", testCase.Output, result)
		}
	}
}
